### ISSUE TITLE
New User without email address fails to load properly

### ISSUE THREAD

**Issue by soggysec** [sourcegraph/sourcegraph#426]
- **Sourcegraph Version: 2.12.0
- **Platform information:** Ubuntu 18.04 - Docker version 18.06.1-ce, build e68fc7a

#### Steps to reproduce:

1. Create a new normal user
2. Don't provide an email address (as it's mentioned optional)
3. Reset user password and login as new user
4. Browse to YourSourceGraph.com/

#### Expected behavior:

Should see search bar and the link bar in the upper right.

#### Actual behavior:

Infinitely spinning wheel under the search box, page never fully loads.

**Comment by soggysec** [comment]
Adding an email address after the fact provides fixes the issue.

**Comment by soggysec** [comment]
there's a graphql query:
```javascript
        query CurrentAuthState {
            currentUser {
                __typename
                id
                sourcegraphID
                username
                avatarURL
                email
                username
                displayName
                siteAdmin
                tags
                url
                organizations {
                    nodes {
                        id
                        name
                    }
                }
                session {
                    canSignOut
                }
                viewerCanAdminister
            }
        }
```
and it fails:
```javascript
{"data":{"currentUser":{"__typename":"User","id":"ABClcjoz","sourcegraphID":3,"username":"joe_user","avatarURL":null,"email":null,"displayName":null,"siteAdmin":true,"tags":[],"url":"/users/joe_user","organizations":{"nodes":[]},"session":{"canSignOut":true},"viewerCanAdminister":true}},"errors":[{"message":"user email not found: [id 3]","path":["currentUser","email"]}]}
```

**Comment by keegancsmith** [comment]
cc @sqs @beyang

**Comment by ijsnow** [comment]
Closed with https://github.com/sourcegraph/sourcegraph/pull/457

**Comment by dadlerj** [comment]
This fix is now available in Sourcegraph 2.12.1 https://docs.sourcegraph.com/admin/install/docker
